base/Enums.jl

```diff
@@ -47,28 +47,34 @@ macro enum(T,syms...)
     if isempty(syms)
         throw(ArgumentError("no arguments given for Enum $T"))
     end
-    if !isa(T,Symbol)
+    basetype = Int32
+    typename = T
+    if isa(T,Expr) && T.head == :(::) && length(T.args) == 2 && isa(T.args[1], Symbol)
+        typename = T.args[1]
+        basetype = eval(current_module(),T.args[2])
+        if !isa(basetype, DataType) || !(basetype <: Integer) || !isbits(basetype)
+            throw(ArgumentError("invalid base type for Enum $typename, $T=::$basetype; base type must be an integer bitstype"))
+        end
+    elseif !isa(T,Symbol)
         throw(ArgumentError("invalid type expression for enum $T"))
     end
-    typename = T
     vals = Array{Tuple{Symbol,Integer}}(0)
     lo = hi = 0
-    i = Int32(-1)
+    i = zero(basetype)
```

Comment:
I switched this from incrementing before a symbol to incrementing afterward because that seemed to be the clearest way to support unsigned base types.